api/v1alpha1/hostedcluster_types.go

```diff
@@ -267,6 +269,51 @@ type ClusterAutoscaling struct {
 	PodPriorityThreshold *int32 `json:"podPriorityThreshold,omitempty"`
 }
 
+// EtcdManagementType is a enum specifying the strategy for managing the clusters etcd instance
+type EtcdManagementType string
+
+const (
+	Managed   EtcdManagementType = "Managed"
+	Unmanaged EtcdManagementType = "Unmanaged"
+)
+
+type EtcdSpec struct {
+	// ManagementType defines how the etcd cluster is managed. Unmanaged means
+	// the etcd cluster is managed by a system outside the hypershift controllers.
+	// Managed means the hypershift controllers manage the provisioning of the etcd cluster
+	// and the operations around it
+	ManagementType EtcdManagementType `json:"managementType"`
+
+	// Managed provides metadata that defines how the hypershift controllers manage the etcd cluster
+	Managed *ManagedEtcdSpec `json:"managed,omitempty"`
+
+	// Unmanaged provides metadata that enables the Openshift controllers to connect to the external etcd cluster
+	Unmanaged *UnmanagedEtcdSpec `json:"unmanaged,omitempty"`
+}
+
+type ManagedEtcdSpec struct {
+
+	//TODO: Ultimately backup policies, etc can be defined here.
+}
+
+type UnmanagedEtcdSpec struct {
+	// Endpoint is the full url to connect to the etcd cluster endpoint. An example is
+	// https://etcd-client:2379
+	Endpoint string `json:"endpoint"`
+
+	// TLS defines a reference to a TLS secret that can be used for client MTLS authentication with
+	// the etcd cluster
+	TLS EtcdTLSConfig `json:"tls"`
```

Comment:
Is this optional?

Reply:
Depends if you want to offer a unauthenticated etcd endpoint (I have never seen a good use case for that as it's a big security risk)

Reply:
I'm asserting no at this point and that it is required. All the components are wired with the expectation that this secret exists as well so I believe it's a valid assumption (downstream components don't support http etcd endpoints)

Reply:
Makes sense to me. The only use case I have in mind would be for testing. However, I will say it's not really any extra work to make it optional in case somebody knows why it could be useful.